R/plot-grid.R

```diff
@@ -132,23 +187,89 @@ plotGrid <- function(plotGridConfiguration) {
 #' patchwork.
 #' @field design Specification of the location of areas in the layout. Can
 #'   either be specified as a text string or by concatenating calls to [area()]
-#'   together. See the examples for further information on use.
+#'   together. See the examples in [wrap_plots()][patchwork::wrap_plots] for
+#'   further information on use.
+#'
+#' @examples
+#'
+#' library(tlf)
+#'
+#' # create a list of plots
+#' ls_plots <- list(
+#'   # first plot
+#'   plotBoxWhisker(mtcars,
+#'     dataMapping = BoxWhiskerDataMapping$new(x = "am", y = "wt"), outliers = FALSE
+#'   ),
+#'   # second plot
+#'   plotBoxWhisker(ToothGrowth,
+#'     dataMapping = BoxWhiskerDataMapping$new(x = "supp", y = "len")
+#'   )
+#' )
+#'
+#' plotGridObj <- PlotGridConfiguration$new(ls_plots)
+#'
+#' # specify further customizations for the plot grid
+#' plotGridObj$title <- "my combined plot"
+#' plotGridObj$subtitle <- "something clever"
+#' plotGridObj$caption <- "my sources"
+#' plotGridObj$nColumns <- 2L
+#' plotGridObj$tagLevels <- "A"
+#' plotGridObj$tagPrefix <- "Plot ("
+#' plotGridObj$tagSuffix <- ")"
+#' plotGridObj$tagColor <- "blue"
+#' plotGridObj$tagSize <- 15
+#' plotGridObj$tagAngle <- 45
+#' plotGridObj$tagPosition <- TagPositions$top
+#' plotGridObj$titleHorizontalJustification <- HorizontalJustification$middle
+#' plotGridObj$subtitleHorizontalJustification <- HorizontalJustification$middle
+#'
+#' # print the object to see its properties
+#' plotGridObj
 #'
 #' @return A `PlotGridConfiguration` object.
 #'
 #' @export
 PlotGridConfiguration <- R6::R6Class(
   "PlotGridConfiguration",
+  inherit = ospsuite.utils::Printable,
   public = list(
     plotList = list(),
+
+    # title ------------------------------------
+
     title = NULL,
+    titleColor = "black",
+    titleSize = 14,
```

Comment:
don't we have this (and other sizes below) as a constant somewhere in the tlf?

Reply:
No, but I have created an enum for it now.